### ISSUE TITLE
support for Visual Studio Mac

### ISSUE THREAD

**Issue by Rohithzr** [Microsoft/nodejstools#1574]
I would guess there is a way to support NodeJS on Visual Studio for Mac?
- Visual Studio Version: Visual Studio Enterprise for Mac v7.0 (Build 3146)

**Comment by paulvanbrenk** [comment]
Thanks for the feedback. This is certainly something we're considering for the future, but for I would like to recommend [Visual Studio Code](https://code.visualstudio.com/) as a great cross-platform alternative.

Are there any specific features you're missing in Visual Studio Code, that we should address?

**Comment by Rohithzr** [comment]
@paulvanbrenk basically I am a fan of Visual Studio as an IDE and last night I started exploring Visual Studio Code. Visual Studio has quite awesome features like Profiling and IntelliSense and I see that IntelliSense is in VS Code too, so I am trying it and will give feedback, but basically I will already use Visual Studio for mac to develop for android, ios and mac so why not node too and when it comes for it why not even go language.

My point, there is no better IDE than VisualStudio hands down and I hope for a day when I can do all my development work on it.

Anyways I hope this feature will come to VisualStudio Mac soon.

Thanks

**Comment by chuks** [comment]
Ditto to @Rohithzr's point. I just downloaded VS for Mac, and wanted to try out the node tools, but realized it is not supported on Mac.

**Comment by paulvanbrenk** [comment]
Thanks for the feedback.

@bowdenk7 taking notes?

**Comment by defionscode** [comment]
Aw man I was stoked for VS until I saw this. I'd be pretty happy to see the mac version support it too.

**Comment by spikyjt** [comment]
VS Code is great for developing entirely in NodeJS but when including npm packages in a C# project either for scripting tools such as Grunt or Gulp or as components of a larger project, it would be great to have these features in the full version of Visual Studio.

**Comment by seryoni** [comment]
I'm encountering performance issues with the Visual Studio Code debugger when working on memory consuming Node.js apps. I was hoping Visual Studio for Mac would solve that but was disappointed to see it's not supported.

**Comment by paulvanbrenk** [comment]
@seryoni Can you open an issue on the [VS Code Node Debugger](https://github.com/Microsoft/vscode-node-debug2/) repo, concerning the performance issues you're seeing?

**Comment by b800h** [comment]
Just quit out of my installation of VS on the Mac when I saw this. 

Add node support and I'm in.

**Comment by fornowatlunch** [comment]
Damn, there is language about support for node all over the Microsoft pages touting VS Community, download the mac version and...no support.

**Comment by paulvanbrenk** [comment]
Thanks for all the comments!

**Comment by Rohithzr** [comment]
agreed @b800h and @fornowatlunch 
I too uninstalled VS for Mac a long time ago, for the specific reason. 

@paulvanbrenk after using VS Code for 6+ months, I can tell you its cons.

- Slow when compared to Visual Studio and Sublime Text (a direct competitor)
- - Opening files
- - Showing Suggestions
- - When using more than one window
- - When using for a long time without closing/quitting
- Less Featured
- - Lint support in not very good (ReactJS)
- - Intellisense doesnt work as greatly as it works on VS (visible difference in speed and usability)
- - lags in scrolling moderately small files

All in all I am tempted to go back to SublimeText, I can't at the moment as I am working on a project and it takes time to switch.

**Comment by paulvanbrenk** [comment]
There is a UserVoice item which is getting much more traction with the VS 4 Mac team. So closing this, so we can combine the vote totals.

Vote here! >>  https://visualstudio.uservoice.com/forums/563332-visual-studio-for-mac

**Comment by sohailhaider** [comment]
Votes on UserVoice been closed, can you please provide support for Node.js for VS on Mac. I love the light-weight features of best IDE in world meanwhile UI for Mac is also awesome.

**Comment by phi1ipp** [comment]
are there any changes to node.js support in coming vs2019 for Mac?

**Comment by Rohithzr** [comment]
@paladique I don't see any possibility.

**Comment by mrdulin** [comment]
@chuks  Yes. I spend a half hour to find where is `Tools > Get Tools and Features...`. 

**VS studio doesn't support nodejs feature on MacOS platform**

**Comment by clepore** [comment]
Here is an issue I submitted to them: https://developercommunity.visualstudio.com/content/problem/579425/using-node-with-nvm-missing-from-vs-path.html

**Comment by anutting** [comment]
Coming back in 2020 still hoping for Node.js support on Visual Studio for Mac. This thread is 3 years old and it's still not implemented.

**Comment by paulvanbrenk** [comment]
> Coming back in 2020 still hoping for Node.js support on Visual Studio for Mac. This thread is 3 years old and it's still not implemented.

I sympathize, but tbh in the last 2 years I've been doing a ton of JS development and VS Code is a great product.

**Comment by svareniem** [comment]
+

**Comment by nosechan** [comment]
> > Coming back in 2020 still hoping for Node.js support on Visual Studio for Mac. This thread is 3 years old and it's still not implemented.
> 
> I sympathize, but tbh in the last 2 years I've been doing a ton of JS development and VS Code is a great product.

I agree VS Code is quite good on JS Development. But the selling point of Visual Studio for Mac is well-supporting on Microsoft Framework,ie .net Core.

**Comment by joystick** [comment]
Hi, 

If you are having troubles with Visual Studio on macOs not being able to find `node` or `npm` while using `nvm` try starting Visual Studio from terminal after ensuring `node -v` and `npm -v` both shows nvm-selected versions, e.g.:
`➜  ~ /Applications/Visual\ Studio.app/Contents/MacOS/VisualStudio &`

For the first time you may be prompted to add dev cert to keychain also:
`➜  ~ dotnet dev-certs https`

**Comment by pbrianmackey** [comment]
For me the command that worked actually was

```
/Applications/Visual\ Studio.app/Contents/MacOS/VisualStudio & dotnet dev-certs https
```

**Comment by dungmv** [comment]
2022 and upvote
